app/code/Magento/SalesRule/Model/Quote/Discount.php

```diff
@@ -281,7 +281,12 @@ private function aggregateDiscountPerRule(
         AddressInterface $address,
         array &$addressDiscountAggregator
     ) {
-        $discountBreakdown = $item->getExtensionAttributes()->getDiscounts();
+        $itemExtAttr = $item->getExtensionAttributes();
+        if ($itemExtAttr && get_class($itemExtAttr) === 'Magento\SalesRule\Api\Data\RuleDiscountInterface') {
```

Comment:
```suggestion
        if ($itemExtAttr && get_class($itemExtAttr) === Magento\SalesRule\Api\Data\RuleDiscountInterface::class) {
```

Reply:
Does it really related to the issue which this PR is going to fix?

Reply:
@swnsma i am not aware of this change. My only changes, as minimum for fixing the issue are in my commit:

https://github.com/magento/magento2/pull/31738/commits/93123bb5d3cf7cece7b9ab8433a321a8645c0e99

Reply:
Also i am not aware of more of these changes you requested from me :)) 
I guess the rest of them came from the rest of the commits with "merge 2.4 develop into this branch"

Reply:
Additional changes added by me. During test coverage, I've found a few more places where added extension attribute could be the reason for the error.

Reply:
Has been updated

Reply:
Hi @engcom-Hotel,
Do these changes cause errors in newly added tests?
Could you please check this moment.
As for me, these changes should be related to the Pull Request.